packages/gensx-openai/src/structured-output.tsx

```diff
@@ -29,70 +32,153 @@ type StructuredOutputProps<O = unknown> = Omit<
 
 type StructuredOutputOutput<T> = T;
 
+// Helper function to create an output schema tool
+const createOutputSchemaTool = <T,>(
+  outputSchema: z.ZodSchema<T>,
+): GSXTool<any> => {
+  // Create a wrapper schema that will contain the output
+  const wrapperSchema = z.object({
+    output: outputSchema,
+  });
+
+  return GSXTool.create({
+    name: "output_schema",
+    description:
+      "When producing a final output, always call this tool to produce a structured output",
```

Comment:
Do you think this description is specific enough? It feels a bit light on details and commands. But I dunno, sometimes less is more for these things.

Reply:
I guess I remember trying to get this to work effectively and it was problematic, depending on the model. 

That being said, here is what I settled on 😆 https://github.com/cortexclick/cortex/blob/a9cee41916697b0a6a8c176c3d957ff81178ece5/cortex-api/content/helpers/createContentExecutors/langgraph/utils/structuredReactAgent.ts#L40